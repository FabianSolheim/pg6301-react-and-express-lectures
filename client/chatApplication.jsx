import React from "react";

export function ChatApplication({ messages }) {
  return (
    <>
      <header>Kristiania Chat</header>
      <main>
        {messages.map(({ message, user }, index) => (
          <div key={index}>
            <strong>{user}:</strong> {message}
          </div>
        ))}
      </main>
      <footer>
        <form>
          <input autoFocus={true} />
          <button>Send</button>
        </form>
      </footer>
    </>
  );
}
