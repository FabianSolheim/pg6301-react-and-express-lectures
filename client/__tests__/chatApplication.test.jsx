import React from "react";
import ReactDOM from "react-dom";
import { ChatApplication } from "../chatApplication";

describe("Chat application", () => {
  it("shows chats", () => {
    const chats = [
      { user: "test user", message: "message 1 " },
      { user: "other user", message: "message 2" },
    ];

    const element = document.createElement("div");
    ReactDOM.render(<ChatApplication messages={chats} />, element);
    expect(element.innerHTML).toMatchSnapshot();
  });
});
