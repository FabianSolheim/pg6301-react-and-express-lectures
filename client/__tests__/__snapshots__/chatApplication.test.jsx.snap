// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chat application shows chats 1`] = `"<header>Kristiania Chat</header><main><div><strong>test user:</strong> message 1 </div><div><strong>other user:</strong> message 2</div></main><footer><form><input value=\\"\\"><button>Send</button></form></footer>"`;
